src/cockpit/389-console/src/lib/ldap_editor/wizards/operations/addUser.jsx

```diff
@@ -39,26 +32,82 @@ class AddUser extends React.Component {
     constructor (props) {
         super(props);
 
-        this.inetorgPersonArray = [
-            'audio', 'businessCategory', 'carLicense', 'departmentNumber', 'displayName', 'employeeNumber',
-            'employeeType', 'givenName', 'homePhone', 'homePostalAddress', 'initials', 'jpegPhoto', 'labeledURI',
-            'mail', 'manager', 'mobile', 'o', 'pager', 'photo', 'roomNumber', 'secretary', 'uid', 'userCertificate',
-            'x500UniqueIdentifier', 'preferredLanguage', 'userSMIMECertificate', 'userPKCS12'
-        ];
-        this.organizationalPersonArray = [
-            'title', 'x121Address', 'registeredAddress', 'destinationIndicator', 'preferredDeliveryMethod', 'telexNumber',
-            'teletexTerminalIdentifier', 'internationalISDNNumber', 'facsimileTelephoneNumber',
-            'street', 'postOfficeBox', 'postalCode', 'postalAddress', 'physicalDeliveryOfficeName', 'ou', 'st', 'l'
-        ];
-        this.personOptionalArray = [
-            'userPassword', 'telephoneNumber', 'seeAlso', 'description'
-        ];
-        this.operationalOptionalArray = ['nsRoleDN'];
-        this.singleValuedAttributes = [
-            'preferredDeliveryMethod', 'displayName', 'employeeNumber', 'preferredLanguage'
-        ];
+        // Objectclasses
+        this.userObjectclasses = {
+            'Basic Account': [
+                'objectclass: top',
+                'objectClass: nsPerson',
+                'objectClass: nsAccount',
+                'objectClass: nsOrgPerson',
+            ],
+            'Posix Account': [
+                'objectclass: top',
+                'objectClass: nsPerson',
+                'objectClass: nsAccount',
+                'objectClass: nsOrgPerson',
+                'objectClass: posixAccount',
+            ],
+            'Service Account': [
+                'objectclass: top',
+                'objectClass: applicationProcess',
+                'objectClass: nsAccount'
+            ],
+        };
 
-        this.requiredAttributes = ['cn', 'sn'];
+        this.allowedAttrs = {
+            'Basic Account': [
+                'businessCategory', 'carLicense', 'departmentNumber',
+                'description', 'employeeNumber', 'employeeType', 'homePhone',
+                'homePostalAddress', 'initials', 'jpegPhoto', 'labeledURI',
+                'legalName', 'mail', 'manager', 'mobile', 'nsRoleDN', 'o',
+                'pager', 'photo', 'preferredLanguage', 'nsCertSubjectDN',
+                'nsSshPublicKey', 'roomNumber', 'seeAlso', 'telephoneNumber',
+                'uid', 'userCertificate', 'userPassword', 'userSMIMECertificate',
+                'userPKCS12', 'x500UniqueIdentifier'
+            ],
+            'PosixAccount': [
```

Comment:
Typo here triggers a TypeError. Posix Account
![Screenshot from 2022-03-07 13-29-51](https://user-images.githubusercontent.com/71395320/157043832-18c4acf3-882b-4bc4-96df-ab1bc04f1fc7.png)

Reply:
Should be fixed now, please review...